proposals/interface-types/Explainer.md

```diff
@@ -413,18 +437,18 @@ using the previously-introduced `string-to-memory` lowering instruction.
 With this 2×2 matrix of {lifting,lowering}×{import,export} covered, we can
 now see a complete example of how one wasm module can [shared-nothing link](#Motivation)
 to another wasm module. In this example, we start with one module providing a
-`set` function which takes a `string` key/value pair:
+`get` function which takes a `string` key and returns the associated value:
 
 ```wasm
 (module
-  (func (export "set_") (param i32 i32 i32 i32) ...)
+  (func (export "get_") (param i32 i32) (result i32 i32) ...)
   ...
-  (@interface func (export "set") (param $key string) (param $val string)
+  (@interface func (export "get") (param $key string) (result string)
     arg.get $key
     string-to-memory "mem" "malloc"
-    arg.get $val
-    string-to-memory "mem" "malloc"
-    call-export "set_"
+    call-export "get_"
+    defer-call-export "free"
+    memory-to-string
   )
 )
```

Comment:
You have elided the memory export. But that is an issue I believe.

Reply:
Thanks, fixed!